packages/core/src/browser/status-bar/status-bar.ts

```diff
@@ -87,6 +80,7 @@ export class StatusBarImpl extends VirtualWidget implements StatusBar {
 
     async setElement(id: string, entry: StatusBarEntry): Promise<void> {
         await this.ready;
+        (entry as any).elementId = id;
```

Comment:
Can we apply elementId to the StatusBarEntry interface to prevent such casting: `(entry as any)`?

Reply:
It has been already reworked